packages/mui-material-next/src/FormLabel/FormLabel.tsx

```diff
@@ -0,0 +1,225 @@
+import PropTypes from 'prop-types';
+import { unstable_composeClasses as composeClasses, useSlotProps } from '@mui/base';
+import { OverridableComponent } from '@mui/types';
+import { unstable_capitalize as capitalize } from '@mui/utils';
+import { useThemeProps, styled } from '../styles';
+import useFormControl from '../FormControl/useFormControl';
+import formLabelClasses, { getFormLabelUtilityClasses } from './formLabelClasses';
+import { FormLabelProps, FormLabelTypeMap, FormLabelOwnerState } from './FormLabel.types';
+
+const useUtilityClasses = (ownerState: FormLabelOwnerState) => {
+  const { classes, color, focused, disabled, error, filled, required } = ownerState;
+  const slots = {
+    root: [
+      'root',
+      `color${capitalize(color)}`,
+      disabled && 'disabled',
+      error && 'error',
+      filled && 'filled',
+      focused && 'focused',
+      required && 'required',
+    ],
+    asterisk: ['asterisk', error && 'error'],
+  };
+
+  return composeClasses(slots, getFormLabelUtilityClasses, classes);
+};
+
+export const FormLabelRoot = styled('label', {
+  name: 'MuiFormLabel',
+  slot: 'Root',
+  overridesResolver: ({ ownerState }, styles) => {
+    return [
+      styles.root,
+      ownerState.color === 'secondary' && styles.colorSecondary,
+      ownerState.filled && styles.filled,
+    ];
+  },
+})<{ ownerState: FormLabelOwnerState }>(({ theme, ownerState }) => {
+  const { vars: tokens } = theme;
+
+  const pxFontSize = tokens.sys.typescale.body.large.size;
+
+  const letterSpacing = `${theme.sys.typescale.body.large.tracking / pxFontSize}rem`;
+
+  return {
+    '--md-comp-form-label-color': tokens.sys.color.secondary,
+    '--md-comp-form-label-font-family': tokens.sys.typescale.body.large.family,
+    '--md-comp-form-label-font-size': theme.typography.pxToRem(pxFontSize), // the pxToRem should be moved to typescale in the future,
+    '--md-comp-form-label-font-weight': tokens.sys.typescale.body.large.weight,
+    '--md-comp-form-label-letter-spacing': letterSpacing,
+    '--md-comp-form-label-line-height': '1.5rem',
+    '--md-comp-form-label-disabled-color': tokens.sys.color.onSurface,
+    '--md-comp-form-label-disabled-opacity': 0.38,
+    '--md-comp-form-label-error-color': tokens.sys.color.error,
+    '--md-comp-form-label-focus-color': tokens.sys.color[ownerState.color],
+    color: 'var(--md-comp-form-label-color)',
+    fontFamily: 'var(--md-comp-form-label-font-family)',
+    fontSize: 'var(--md-comp-form-label-font-size)',
+    fontWeight: 'var(--md-comp-form-label-font-weight)',
+    lineHeight: 'var(--md-comp-form-label-line-height)',
+    padding: 0,
+    position: 'relative',
+    [`&.${formLabelClasses.focused}`]: {
+      color: 'var(--md-comp-form-label-focus-color)',
+    },
+    [`&.${formLabelClasses.disabled}`]: {
+      color:
+        'color-mix(in srgb, var(--md-comp-form-label-disabled-color), transparent calc(var(--md-comp-form-label-disabled-opacity) * 100%))',
```

Comment:
From: https://una.im/color-mix-opacity/
A neat trick using `color-mix` to adjust the opacity of `color` ✨

Reply:
👏🏼 also using this [in the Chip component](https://github.com/mui/material-ui/blob/master/packages/mui-material-next/src/Chip/Chip.tsx#L253). It's very useful for Material You's styles.